astrokat/observe_main.py

```diff
@@ -365,134 +403,158 @@ def run_observation(opts, kat):
+                    if obs_cntr < nr_obs_loops - 1:
+                        user_logger.info(log_msg)
+                    else:
+                        user_logger.error(log_msg)
                     continue
             else:
-                # else assume rollover at midnight to next day
-                out_range = (ephem.hours(local_lst) < ephem.hours(str(start_lst))) and (
-                    ephem.hours(local_lst) > ephem.hours(str(end_lst))
+                # lst ends after midnight
+                if _next_day(start_lst, end_lst, local_lst):
+                    if obs_cntr < nr_obs_loops - 1:
+                        user_logger.info(log_msg)
+                    else:
+                        user_logger.error(log_msg)
+                    continue
+
+            # Verify that it is worth while continuing with the observation
+            # The filter functions uses the current time as timestamps
+            # and thus incorrectly set the simulation timestamp
+            if not kat.array.dry_run:
+                # Quit early if there are no sources to observe
+                if len(catalogue.filter(el_limit_deg=opts.horizon)) == 0:
+                    raise NoTargetsUpError(
+                        "No targets are currently visible - "
+                        "please re-run the script later"
+                    )
+                # Quit early if the observation requires all targets to be visible
+                if opts.all_up and (
+                    len(catalogue.filter(el_limit_deg=opts.horizon)) != len(catalogue)
+                ):
+                    raise NotAllTargetsUpError(
+                        "Not all targets are currently visible - please re-run the script"
+                        "with --visibility for information"
+                    )
+
+            # List sources and their associated functions from observation tags
+            not_cals_filter_list = []
+            for cal_type in cal_tags:
+                not_cals_filter_list.append("~{}".format(cal_type))
+                cal_array = [cal.name for cal in catalogue.filter(cal_type)]
+                if len(cal_array) < 1:
+                    continue  # do not display empty tags
+                user_logger.info(
+                    "{} calibrators are {}".format(str.upper(cal_type[:-3]), cal_array)
                 )
-                if out_range:
-                    user_logger.error(
-                        "Local LST outside LST range "
-                        "{}-{}".format(
-                            ephem.hours(str(start_lst)), ephem.hours(str(end_lst))
-                        )
+            user_logger.info(
+                "Observation targets are [{}]".format(
+                    ", ".join(
+                        [
+                            repr(target.name)
+                            for target in catalogue.filter(not_cals_filter_list)
+                        ]
```

Comment:
Perhaps, simplify this by assigning this logic to a var and then calling the var here... this is a bit harder to read.